core/store/tx_manager.go

```diff
@@ -714,23 +721,61 @@ func (txm *EthTxManager) handleSafe(
 	return nil
 }
 
-// bumpGas creates a new transaction attempt with an increased gas cost
+// BumpGasByIncrement returns a new gas price increased by the larger of either
+// a percentage bump or a fixed size bump
+func (txm *EthTxManager) BumpGasByIncrement(originalGasPrice *big.Int) *big.Int {
+	// Similar logic is used in geth
+	// See: https://github.com/ethereum/go-ethereum/blob/8d7aa9078f8a94c2c10b1d11e04242df0ea91e5b/core/tx_list.go#L255
+	// And: https://github.com/ethereum/go-ethereum/blob/8d7aa9078f8a94c2c10b1d11e04242df0ea91e5b/core/tx_pool.go#L171
+	minimumGasBumpByPercentage := new(big.Int).Div(
+		new(big.Int).Mul(
+			originalGasPrice,
+			big.NewInt(100+int64(txm.config.EthGasBumpPercent())),
+		),
+		big.NewInt(100),
+	)
+	minimumGasBumpByIncrement := new(big.Int).Add(originalGasPrice, txm.config.EthGasBumpWei())
+	if minimumGasBumpByIncrement.Cmp(minimumGasBumpByPercentage) < 0 {
+		return minimumGasBumpByPercentage
+	}
+	return minimumGasBumpByIncrement
+}
```

Comment:
can use short term 'min' and 'perc instead of the long form.

Reply:
What do you think about making this a function instead of a method?  It doesn't have to understand stuff about config etc and does not modify the tx manager.

Reply:
I did originally write it as a function but then you have to pass in the config every time it's called. It turned out simpler to write and test to have it this way.